swan-lake/integration-technology-tools/edi-tool.md

```diff
@@ -6,4 +6,349 @@ keywords: ballerina, programming language, edi, contract
+
+## Compatibility
+
+|                            | Compatible with |
+|----------------------------|------------------|
+| Ballerina language version | 2201.7.0         |
+| Java Development Kit (JDK) | 11               |
+
+## Usage
+
+The usage of the tool is described below.
+
+```
+bal edi codegen <EDI schema path> <output path>
+```
+
+The above command generates all Ballerina records and parsing functions required for working with data in the given EDI schema and writes those into the file specified in the output path. The generated parsing function (i.e., `fromEdiString(...)`) can read EDI text files into generated records, which can be accessed from the Ballerina code similar to accessing any other Ballerina record. Similarly, the generated serialization function (i.e., `toEdiString(...)`) can serialize generated Ballerina records into EDI text.
+
+## Example
+
+A simple EDI schema (`edi-schema.json` file) is shown below.
+
+```json
+{
+    "name": "SimpleOrder",
+    "delimiters" : {"segment" : "~", "field" : "*"},
+    "segments" : {
+        "HDR": {
+            "tag" : "header",
+            "fields" : [{"tag" : "code"}, {"tag" : "orderId"}, {"tag" : "organization"}, {"tag" : "date"}]
+        },
+        "ITM": {
+            "tag" : "items",
+            "maxOccurances" : -1,
+            "fields" : [{"tag" : "code"}, {"tag" : "item"}, {"tag" : "quantity", "dataType" : "int"}]
+        }
+    }
+}
+```
+
+The above schema can be used to parse EDI documents with one HDR segment (mapped to `header`) and any number of ITM segments (mapped to `items`). The HDR segment contains three fields, which are mapped to `orderId`, `organization`, and `date`. Each ITM segment contains two fields mapped to `item` and `quantity`. Below is a sample EDI document that can be parsed using the above schema (`edi-sample1.edi` file).
+
+```
+HDR*ORDER_1201*ABC_Store*2008-01-01~
+ITM*A-250*12
+ITM*A-45*100
+ITM*D-10*58
+ITM*K-80*250
+ITM*T-46*28
+```
+
+### Generate Ballerina records
+
+Ballerina records for the EDI schema in the `edi-schema.json` file can be generated as follows (the generated Ballerina records will be saved in `orderRecords.bal`).
+
+```
+bal edi codegen -s resources/edi-schema1.json -o modules/hmartOrder/orderRecords.bal
+```
+
+The Ballerina records generated for the above schema are shown below.
+
+```ballerina
+type Header_Type record {|
+   string code?;
+   string orderId?;
+   string organization?;
+   string date?;
+|};
+
+type Items_Type record {|
+   string code?;
+   string item?;
+   int quantity?;
+|};
+
+type SimpleOrder record {|
+   Header_Type header;
+   Items_Type[] items?;
+|};
+```
+
+### Read EDI files
+
+The generated `fromEdiString` function can be used to read EDI text files into the generated Ballerina record as shown below. 
```

Comment:
Don't we need a comma before `as shown below`. i.e. `Ballerina record, as shown below.`

Reply:
Added.